homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -109,6 +112,18 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
     await gather(
         *[coordinator.async_refresh() for coordinator in coordinators_by_id.values()]
     )
+
+    list_devices_service = make_list_devices(api)
+    hass.services.async_register(
+        DOMAIN,
+        "list_devices",
+        list_devices_service,
+        supports_response=SupportsResponse.ONLY,
+    )
+
+    send_command_service = make_send_command_service(api)
+    hass.services.async_register(DOMAIN, "send_command", send_command_service)
```

Comment:
We should not create a send command service as that leaves the user without context. It's a waaay better user experience if we can contextualize the calls. So for example if you would use this service to activate something, rather create a service to do that activation via that. Or use entities to accomodate that action

Reply:
I am sorry, I am not sure I understood. Are you proposing to add a service to for instance a plug to add additional commands ?
I made the command service to be able to control devices difficult to adapt to Home Assistant like a TV (the interfaces were sto different that made me puzzled) and other non supported devices.
I agree, making entities would be easier to use  though.